internal/auth/client_kerberos_factory.go

```diff
@@ -0,0 +1,57 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/containerssh/libcontainerssh/config"
+	"github.com/containerssh/libcontainerssh/http"
+	"github.com/containerssh/libcontainerssh/internal/metrics"
+	"github.com/containerssh/libcontainerssh/log"
+
+	"github.com/jcmturner/gokrb5/v8/keytab"
+	krb5cfg "github.com/jcmturner/gokrb5/v8/config"
+)
+
+func NewKerberosClient(
+	cfg config.AuthConfig,
+	logger log.Logger,
+	metrics metrics.Collector,
+) (Client, error) {
+
+	if cfg.Method != config.AuthMethodKerberos {
+		return nil, fmt.Errorf("authentication is not set to kerberos")
+	}
+
+	if err := cfg.Validate(); err != nil {
+		return nil, err
+	}
+
+	kt, err := keytab.Load(cfg.Kerberos.Keytab)
+	if err != nil {
+		return nil, err
```

Comment:
Please wrap this error for user friendliness and add the path to the keytab file to the error message.